### PR TITLE
Compile on OCaml 4.06

### PR DESCRIPTION
**Pull request by vasilisp**
The previous OASIS-generated `setup.ml` doesn't work on OCaml 4.06 (`-safe-string`). Additionally, it is time to migrate to PPX :). Finally, I added an `opam` file to facilitate pinning the package.

### CODE REVIEW

**Comment by hannesm** [comment]
thanks, it looks to me that @avsm or @samoht should have an opinion here... it may also be worth to do a release with these changes in (and maybe switching to a modern build system instead of oasis).

**Comment by samoht** [comment]
That looks great, thanks! Indeed switching to a more modern build system is also welcome :-) I'll try to merge these patch and release a new version shortly.

File: opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+]
+remove: [
+  ["ocamlfind" "remove" "lazy-trie"]
+]
+build-doc: [ "ocaml" "setup.ml" "-doc" ]
+depends: [
+  "ocamlfind"
+  "sexplib"
+  "type_conv"
+  "camlp4"
```
Comment (hcarty) [inline opam]: `type_conv` and `campl4` likely need to be replaced with the appropriate ppx here.

Comment (vasilisp) [reply opam]: Right, fixed.

**Pull request merged**